docs: Clarification About PipelineRun YAML Validation Error

zakisk opened the pull request:
removed a statement from docs that wrongly describing about trigerring PipelineRuns if one of the PipelineRun yaml is invalid.

# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

<!-- Describe your changes here- ideally you can get that description straight from your descriptive commit message(s)! -->

# Submitter Checklist

- [x] 📝 Ensure your commit message is clear and informative. Refer to the How to write a git commit message guide. Include the commit message in the PR body rather than linking to an external site (e.g., Jira ticket).

- [x] ♽ Run make test lint before submitting a PR to avoid unnecessary CI processing. Consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the repository root for an efficient workflow.

- [x] ✨ We use linters to maintain clean and consistent code. Run make lint before submitting a PR. Some linters offer a --fix mode, executable with make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) are installed).

- [x] 📖 Document any user-facing features or changes in behavior.

zakisk commented:
/test

chmouel commented:
that's a bug if it doesn't work like that.... what make you think there is a defect?

zakisk commented:
> what make you think there is a defect?

first I saw this on [docs PR](https://github.com/openshift/openshift-docs/pull/94302), and I suggested author to remove the statement after seeing source code and I reproduced the behavior locally as well

zakisk commented:
@chmouel so you mean it's a bug? do you want me to fix?? instead of deleting the statement from docs??

chmouel commented:
I was pretty sure i have implemented https://issues.redhat.com/browse/SRVKP-6877 (but didn't update the status and closed it)...

this is why we have yaml syntax error reporting, because before we would just fail on the first yaml lint errors and report it to the check interface, when i have fixed 6877 we didn't report anymore those yaml lint errors so we had to show them as comment on the PullRequest....

I have invesitaged a bit and it indeed not parsing the successfull one when there is a yaml syntax error
It does continue properly when there is yaml apply errors (when the controller reject it for example) but should report that error on comment. I have logged this jira here for that bug:

https://issues.redhat.com/browse/SRVKP-7894

zakisk commented:
> I was pretty sure i have implemented https://issues.redhat.com/browse/SRVKP-6877 (but didn't update the status and closed it)...
> 
> this is why we have yaml syntax error reporting, because before we would just fail on the first yaml lint errors and report it to the check interface, when i have fixed 6877 we didn't report anymore those yaml lint errors so we had to show them as comment on the PullRequest....
> 
> I have invesitaged a bit and it indeed not parsing the successfull one when there is a yaml syntax error It does continue properly when there is yaml apply errors (when the controller reject it for example) but should report that error on comment. I have logged this jira here for that bug:
> 
> https://issues.redhat.com/browse/SRVKP-7894

So if one PipelineRun has invalid YAML template others should be triggered normally and only bad one is left with reporting on pull request as comment, right??

chmouel commented:
@zakisk 

> So if one PipelineRun has invalid YAML template others should be triggered normally and only bad one is left with reporting on pull request as comment, right??

correct

zakisk committed:
dcos: clarification about YAML validation error

added a clarification about PR comment on PipelineRun
validation errors.

Signed-off-by: Zaki Shaikh <zashaikh@redhat.com>

zakisk commented:
@chmouel rephrased the docs. PTAL
